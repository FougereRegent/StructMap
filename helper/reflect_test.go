package helper

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

type gender int

const (
	Male gender = iota
	Female
)

func TestGetPropertiesInt(t *testing.T) {
	type typeStructIntTest struct {
		Field1    int
		Field2    int8
		Field3    int32
		Field4    int64
		PtrField1 *int
		PtrField2 *int8
		PtrField3 *int32
		PtrField4 *int64
	}

	result := GetProperties(typeStructIntTest{})

	snaps.MatchJSON(t, result)
}

func TestGetPropertiesFloat(t *testing.T) {
	type typeStructFloatTest struct {
		Field1    float32
		Field2    float64
		PtrField1 *float32
		PtrField2 *float64
	}

	result := GetProperties(typeStructFloatTest{})

	snaps.MatchJSON(t, result)
}

func TestGetPropertiesString(t *testing.T) {
	type typeStructStringTest struct {
		Field1    string
		PtrField1 *string
	}

	result := GetProperties(typeStructStringTest{})

	snaps.MatchJSON(t, result)
}

func TestGetPropertiesBoolean(t *testing.T) {
	type typeStructBooleanTest struct {
		Field1    bool
		PtrField1 *bool
	}

	result := GetProperties(typeStructBooleanTest{})

	snaps.MatchJSON(t, result)
}

func TestGetPropertiesTags(t *testing.T) {
	type userStructTest struct {
		FirstName string `structmap:"FirstName"`
		LastName  string `structmap:"LastName" json:"Test"`
		Age       int
		Address   struct {
			ZipCode int16
			Country string
			Ciry    string
		} `structmap:"Address"`
		Gender gender
	}

	result := GetProperties(userStructTest{})

	snaps.MatchJSON(t, result)
}
