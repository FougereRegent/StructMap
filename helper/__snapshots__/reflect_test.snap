
[TestGetPropertiesInt - 1]
[
 {
  "Name": "Field1",
  "Tag": "",
  "Type": 2
 },
 {
  "Name": "Field2",
  "Tag": "",
  "Type": 3
 },
 {
  "Name": "Field3",
  "Tag": "",
  "Type": 5
 },
 {
  "Name": "Field4",
  "Tag": "",
  "Type": 6
 },
 {
  "Name": "PtrField1",
  "Tag": "",
  "Type": 22
 },
 {
  "Name": "PtrField2",
  "Tag": "",
  "Type": 22
 },
 {
  "Name": "PtrField3",
  "Tag": "",
  "Type": 22
 },
 {
  "Name": "PtrField4",
  "Tag": "",
  "Type": 22
 }
]
---

[TestGetPropertiesFloat - 1]
[
 {
  "Name": "Field1",
  "Tag": "",
  "Type": 13
 },
 {
  "Name": "Field2",
  "Tag": "",
  "Type": 14
 },
 {
  "Name": "PtrField1",
  "Tag": "",
  "Type": 22
 },
 {
  "Name": "PtrField2",
  "Tag": "",
  "Type": 22
 }
]
---

[TestGetPropertiesString - 1]
[
 {
  "Name": "Field1",
  "Tag": "",
  "Type": 24
 },
 {
  "Name": "PtrField1",
  "Tag": "",
  "Type": 22
 }
]
---

[TestGetProperties - 1]
[
 {
  "Name": "Field1",
  "Tag": "",
  "Type": 1
 },
 {
  "Name": "PtrField1",
  "Tag": "",
  "Type": 22
 }
]
---

[TestGetPropertiesBoolean - 1]
[
 {
  "Name": "Field1",
  "Tag": "",
  "Type": 1
 },
 {
  "Name": "PtrField1",
  "Tag": "",
  "Type": 22
 }
]
---

[TestGetPropertiesTags - 1]
[
 {
  "Name": "FirstName",
  "Tag": "FirstName",
  "Type": 24
 },
 {
  "Name": "LastName",
  "Tag": "LastName",
  "Type": 24
 },
 {
  "Name": "Age",
  "Tag": "",
  "Type": 2
 },
 {
  "Name": "Address",
  "Tag": "Address",
  "Type": 25
 },
 {
  "Name": "Gender",
  "Tag": "",
  "Type": 2
 }
]
---
